sonic_platform_base/chassis_base.py

```diff
@@ -0,0 +1,94 @@
+#
+# chassis_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a chassis device in SONiC.
+#
+
+try:
+    import abc
+    import sys
+    from . import device_base
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+
+class ChassisBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a platform chassis
+    """
+
+    __metaclass__ = abc.ABCMeta
```

Comment:
This `__metaclass__` syntax is python2 compliant but will not do anything for python3.
With this syntax instantiating ChassisBase will fail as expected in python2 but will succeed in python3.
If this compatibility matters to you, the six module will have to be used since the syntax are different in both language.

Also `__metaclass__` is already defined in `device_base.DeviceBase`

Reply:
Thanks for bringing this up, as this library will most likely be imported into both Python 2 and Python 3 code. I will address this in my next commit.